File: TopCoder/src/main/java/com/leetcode/array/leetcode/UncommonWordsFromTwoSentences.java
package com.leetcode.array.leetcode;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;
import java.util.function.Consumer;
import java.util.function.Function;
import java.util.stream.Collectors;

public class UncommonWordsFromTwoSentences {
    public String[] uncommonFromSentences(String A, String B) {

        Map<String, Long> map = Arrays.asList(A.split(" ")).stream()
                .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
        Consumer<String> consumer = value -> {
            if (!map.containsKey(value)) {
                map.put(value, 1L);
            } else {
                map.put(value, map.get(value) + 1);
            }
        };

        Arrays.asList(B.split(" ")).stream().forEach(consumer);

        Set<String> set = map.entrySet().stream()
                .filter( stringLongEntry -> stringLongEntry.getValue() == 1L).map(Map.Entry::getKey).
                        collect(Collectors.toSet());
        return (String[])set.toArray(new String[0]);
    }


    public static void main(String[] args) {
        String A = "apple apple";
        String B = "banana";
        String[] res = new UncommonWordsFromTwoSentences().uncommonFromSentences(A, B);
        Arrays.stream(res).forEach(System.out::print);
    }
}

File: TopCoder/src/main/java/com/leetcode/array/leetcode/UncommonWordsFromTwoSentences.py

class Solution:
    """
    def uncommonFromSentences(self, A, B):
        dict = {}

        arr = A.split(' ')

        for str in arr:
            dict[str] = dict.get(str, 0) + 1

        arr = B.split(' ')
        for str in arr:
            dict[str] = dict.get(str, 0) + 1

        result = [k for k, v in dict.items() if v == 1]
        return result
    """
    def uncommonFromSentences(self, A, B):
        da, db = {}, {}

        for wa in A.split(' '):
            da[wa] = da.get(wa, 0) + 1

        for wb in B.split(' '):
            db[wb] = db.get(wb, 0) + 1

        return [a for a in da if da[a] == 1 and (a not in db)] + [b for b in db if db[b] == 1 and (b not in da)]

A = 'this apple is sweet'
B = 'this apple is sour'

sol = Solution()
res = sol.uncommonFromSentences(A, B)
print(res)